src/Api/ProjectsApi.php

```diff
@@ -328,8 +328,16 @@ public function projectIdReportPost(string $id, ProjectReportRequest $project_re
    */
   public function projectIdDelete(string $id, &$responseCode, array &$responseHeaders)
   {
-    // TODO: Implement projectIdDelete() method.
-    $responseCode = Response::HTTP_NOT_IMPLEMENTED;
+    $user = $this->facade->getAuthenticationManager()->getUserFromAuthenticationToken($this->getAuthenticationToken());
+    if (is_null($user)) {
+      $responseCode = Response::HTTP_UNAUTHORIZED;
+
+      return null;
+    }
+
+    $success = $this->facade->getProcessor()->deleteProjectById($id, $user);
+
+    $responseCode = $success ? Response::HTTP_NO_CONTENT : Response::HTTP_BAD_REQUEST;
```

Comment:
why bad request?

404 would sound better to me; if there is an issue in the specification please fix it.